src/_posts/2016-12-23-sonarqube.md

```diff
@@ -0,0 +1,147 @@
+
+## Create a new project
+
+In my version of Visual Studio (Community 2015), you can do this by clicking on *"File - New - Project"* on the main menu, then *"Class Library"* from *"Templates - Visual C#"*. Give it a interesting name, which I will assume to be **YourProjectName** for the rest of this post.
+
+Add some code that has some quality issues (eg a variable that is declared but never used). You can use the [the full list of SonarQube C# issues](http://dist.sonarsource.com/reports/coverage/rules/csharpsquid_rules_coverage.html) for inspiration. Alternatively you can copy and paste [some of mine](https://github.com/ceddlyburge/sonarqube-nemo-on-appveyor/blob/master/ExampleSonarQubeIssues.cs).
+
+Install the [SonarLint Visual Studio Plugin](https://marketplace.visualstudio.com/items?itemName=SonarSource.SonarLintforVisualStudio). This highlights quality issues in your code as you type and gives you a chance to fix them before comitting.
+
+## Integrate with AppVeyor
+
+You will need to link an AppVeyor account to your GitHub one, so let's do that:
+
+* Navigate to your repo in GitHub
+* Click "Settings" on the repo
+* Click "Integrations and services"
+* Click "Browse Directory"
+* Click "AppVeyor"
+* Click "Configure"
+* Click "Grant Access"
+
+Now Log in to [AppVeyor.com](https://ci.appveyor.com), probably using your GitHub account
+
+* Click "Projects"
+* Click "New Project"
+* Choose your GitHub repository and click "Add"
+
+## Sign up with SonarQube and generate an Authentication Token
+
+* Create an account at [sonarqube.com/sessions/new](https://sonarqube.com/sessions/new)
+* Click on [sonarqube.com/account/security/](https://sonarqube.com/account/security/)
+* Enter a token name and click "Generate"
+* Make a note of the generated token (**YourSonarQubeToken** from now on)
+
+## Run SonarQube Analysis Locally
+
+When working with appveyor, it always makes sense to test on your own computer first. The feedback is immediate and you iterate very quickly. It takes a lot longer to modify the appveyor.yml file, push it and wait for a build to go through. Also, if it works locally but doesn't work on AppVeyor, you know the problem is a configuration difference between your computer and the AppVeyor environment (eg a different version of msbuild).
+
+Instead of committing SonarQube executables to the repo, we will download them during the build using Chocolatey.
+
+### Install chocolatey
+
+* [Install Chocolatey](https://chocolatey.org/install) from an administrator command prompt / powershell.
+* Close the command prompt
+
+### Install SonarQube MSBuild Runner
+
+* Open a new administrator command prompt / powershell.
+* ```choco install "msbuild-sonarqube-runner" -y```
+
+### Analyse and upload to SonarQube
+
+* `MSBuild.SonarQube.Runner.exe begin /k:"**YourUniqueProjectName**" /d:"sonar.host.url=https://sonarqube.com" /d:"sonar.login=**YourSonarQubeToken**"`
```

Comment:
Hi, I've done the first two items, but I'm less keen on this one. I did have it as a code block 
originally but it was hard to read, so I changed it to single line bullet points and it scanned a lot better. 

If you insist I'll do it of course, but I'm not keen.

Cheers

Cedd